test/nextjs.test.js

```diff
@@ -976,16 +976,213 @@ describe(`Next.js App Router: ${printPath("[test/nextjs.test.js]")}`, function (
         const requestWhereHandlerThrowsUnknownError = authenticatedRequest;
         const unknownError = new Error("Unknown error");
         try {
-            await withSession(requestWhereHandlerThrowsUnknownError, async (err, session) => {
-                throw unknownError;
-            });
+            await withPreParsedRequestResponse(
+                requestWhereHandlerThrowsUnknownError,
+                async (baseRequest, baseResponse) => {
+                    throw unknownError;
+                }
+            );
             assert.fail("should not come here");
         } catch (error) {
             assert.strictEqual(error, unknownError);
         }
     });
 });
 
+describe(`getSSRSession:hasToken`, function () {
+    describe("tokenTransferMethod = any", function () {
+        before(async function () {
+            process.env.user = undefined;
+            await killAllST();
+            await setupST();
+            const connectionURI = await startST();
+            ProcessState.getInstance().reset();
+            SuperTokens.init({
+                supertokens: {
+                    connectionURI,
+                },
+                appInfo: {
+                    apiDomain: "api.supertokens.io",
+                    appName: "SuperTokens",
+                    apiBasePath: "/api/auth",
+                    websiteDomain: "supertokens.io",
+                },
+                recipeList: [
+                    EmailPassword.init(),
+                    Session.init({
+                        getTokenTransferMethod: () => "any",
+                    }),
+                ],
+            });
+        });
+
+        after(async function () {
+            await killAllST();
+            await cleanST();
+        });
+
+        it("should return hasToken value correctly", async function () {
+            const requestWithNoToken = new NextRequest("http://localhost:3000/api/get-user");
+
+            sessionContainer = await getSSRSession(requestWithNoToken.cookies.getAll(), requestWithNoToken.headers);
+
+            assert.equal(sessionContainer.hasToken, false);
+
+            const requestWithTokenInHeader = new NextRequest("http://localhost:3000/api/get-user", {
+                headers: {
+                    Authorization: `Bearer some-random-token`,
```

Comment:
on second thought, hasToken should only be true if some token is found in the request (based on the token transfer method) AND if that token is actually one that was issued by ST. Cause if it's a random other token, then we want hasToken: false, so that we don't attempt refreshing unnecessarily. 

In order to check if the token was actually issued by ST, see how it's done here: https://github.com/supertokens/supertokens-node/blob/master/lib/ts/recipe/session/recipeImplementation.ts#L174

Reply:
done!